diff --git a/ext/sass/Rakefile b/ext/sass/Rakefile
index 77ced01..1e60ab0 100644
--- a/ext/sass/Rakefile
+++ b/ext/sass/Rakefile
@@ -18,9 +18,7 @@ file 'protoc.exe' do |t|
 end
 
 file 'dart-sass' do |t|
-  archive = fetch(ENV.fetch(t.name.tr('-', '_').upcase) { Configuration.default_dart_sass })
-  unarchive archive
-  rm archive
+  symlink(ENV.fetch(t.name.tr('-', '_').upcase), 'dart-sass')
 end
 
 file 'cli.rb' => %w[dart-sass] do |t|
